WiseUnpacker/Files/MultiPartFile.cs

```diff
@@ -147,16 +148,19 @@ public override int Read(byte[] buffer, int offset, int count)
             }
             else
             {
-                byte[] buf = new byte[0xffff];
+                byte[] buf = new byte[count];
                 bufpos = 0;
                 do
                 {
+                    if (mf == null)
+                        break;
```

Comment:
Can you add this at Line 140 as well? There's a chance that `mf` is null at that point and it will cause a similar issue. You just need to return `0` immediately at that point.

Reply:
Let me be more specific:

```c#
if (Position > mf._partEnd)
    return 0;
```

can become

```c#
if (mf == null || Position > mf._partEnd)
    return 0;
```